src/iterative_sorting/iterative_sorting.py

```diff
@@ -1,27 +1,39 @@
 # TO-DO: Complete the selection_sort() function below
 def selection_sort(arr):
     # loop through n-1 elements
-    for i in range(0, len(arr) - 1):
-        cur_index = i
-        smallest_index = cur_index
-        # TO-DO: find next smallest element
-        # (hint, can do in 3 loc)
-        # Your code here
-
+    for base in range(0, len(arr) - 1):
+        for i in range(0, len(arr) - base):
+            cur_index = i + base
+            smallest_index = arr[base]
+            # TO-DO: find next smallest element
+            # (hint, can do in 3 loc)
+            # Your code here
+            if arr[cur_index] < smallest_index:
+                smallest_index = arr[cur_index]
+                arr[cur_index] = arr[base]
+                arr[base] = smallest_index
+        base += 1
 
         # TO-DO: swap
         # Your code here
-
     return arr
 
 
 # TO-DO:  implement the Bubble Sort function below
 def bubble_sort(arr):
     # Your code here
+    for base in range(0, len(arr) - 1):
+        for i in range(1, len(arr) - base):
+            if arr[i] < arr[i - 1]:
+                smallest = arr[i]
+                arr[i] = arr[i - 1]
+                arr[i - 1] = smallest
 
+        base += 1
 
     return arr
```

Comment:
Try to optimize the bubble sort.
What if the given array is almost sorted? Does it have to go through the nested for loop every time?